src/store/components/CollectionItemCard/index.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+
+import './style.less';
+import { useHistory } from 'react-router';
+
+type Uuid = string;
+
+interface MerchandiseCollectionModel {
+  uuid: Uuid;
+  title: string;
+  color: string; // hex color used for the title (each collection title has its own text color)
+  description: string;
+  archived: boolean;
+  items: MerchandiseItemModel[];
+}
+
+interface MerchandiseItemModel {
+  uuid: Uuid;
+  itemName: string;
+  picture: string;
+  description: string;
+  hidden: boolean;
+  // then is set to true, otherwise false
+  options: MerchandiseItemOptionModelProps[];
+}
+
+interface MerchandiseItemOptionModelProps {
+  quantity: number;
+  price: number;
+  discountPercentage: number;
+  type: string; // e.g. 'size', 'shape'
+  value: string; // e.g. 'S', 'M', 'L' if this.type === 'size'
+  position: number; // e.g. 0, 1, 2 (for sort order, i.e. XS < S < M < L < XL etc)
+}
+
+const CollectionItemCard: React.FC<MerchandiseItemModel> = (props: MerchandiseItemModel) => {
+  const { uuid, itemName, description, hidden, options, picture } = props;
+
+  const history = useHistory();
+
+  if (hidden) {
+    return null;
+  }
+
+  const outOfStock = options.every((option) => option.quantity === 0);
+  const onSale = options.some((option) => option.discountPercentage !== 0);
+  const priceRange = options.reduce(
+    (acc, option) => {
+      const newAcc = acc;
+      if (acc.low > option.price) {
+        newAcc.low = option.price;
+      } else if (acc.high < option.price) {
```

Comment:
This line should check if the `option.price` is higher **and** lower than the current ranges. Otherwise, single option items don't have a correct `high` value.

```suggestion
      }
      if (acc.high < option.price) {
```

Reply:
Nice catch. My bad.